### PR TITLE
[IOTDB-5627] Optimize Schedule related metrics and Update Performance Overview Dashboard

### PR DESCRIPTION
**Pull request by SpriCoder**
1. Add Local/Remote Schedule Metrics
2. Update Related Docs
3. Update the Performance Overview Dashboard

image

### CODE REVIEW

**Comment by SpriCoder** [comment]
image
image

**Comment by SpriCoder** [comment]
image

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/scheduler/AsyncPlanNodeSender.java [hunk 1]
```diff
@@ -79,6 +82,8 @@ public void sendAll() {
 
   public void waitUntilCompleted() throws InterruptedException {
     countDownLatch.await();
+    PerformanceOverviewMetricsManager.getInstance()
```
Comment (OneSizeFitsQuorum) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/scheduler/AsyncPlanNodeSender.java:85]: If you record it this way, RemoteCost is always longer than the localCost

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/ApplicationStateMachineProxy.java [hunk 2]
```diff
@@ -132,7 +136,17 @@ public CompletableFuture<Message> applyTransaction(TransactionContext trx) {
         }
         IConsensusRequest deserializedRequest =
             applicationStateMachine.deserializeRequest(applicationRequest);
+
+        long startWriteTime = System.nanoTime();
         TSStatus result = applicationStateMachine.write(deserializedRequest);
+        MetricService.getInstance()
```
Comment (OneSizeFitsQuorum) [inline consensus/src/main/java/org/apache/iotdb/consensus/ratis/ApplicationStateMachineProxy.java:142]: we need to distinguish between local and remote apply to make sense of recording only the former

File: consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java [hunk 3]
```diff
@@ -221,7 +221,17 @@ public TSStatus write(IConsensusRequest request) {
...
+      long startWriteTime = System.nanoTime();
       TSStatus result = stateMachine.write(planNode);
+      MetricService.getInstance()
+          .timer(
+              System.nanoTime() - startWriteTime,
+              TimeUnit.NANOSECONDS,
+              Metric.PERFORMANCE_OVERVIEW_SCHEDULE_DETAIL.toString(),
+              MetricLevel.IMPORTANT,
+              Tag.STAGE.toString(),
+              "stateMachine");
```
Comment (OneSizeFitsQuorum) [inline consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java:233]: `PerformanceOverviewMetrics.STATEMACHINE`?

File: node-commons/src/main/java/org/apache/iotdb/commons/service/metric/enums/Metric.java [hunk 4]
```diff
@@ -86,6 +86,7 @@ public enum Metric {
   DRIVER_SCHEDULER,
   PERFORMANCE_OVERVIEW,
   PERFORMANCE_OVERVIEW_DETAIL,
+  PERFORMANCE_OVERVIEW_SCHEDULE_OVERVIEW,
   PERFORMANCE_OVERVIEW_SCHEDULE_DETAIL,
```
Comment (OneSizeFitsQuorum) [inline node-commons/src/main/java/org/apache/iotdb/commons/service/metric/enums/Metric.java:90]: do we need a `PERFORMANCE_OVERVIEW_STATEMECHINE_DETAIL` for those operations in dataRegion?

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**